### PR TITLE
wip: Ability to test frontend

### PR DESCRIPTION
**Pull request by jparkrr**
WIP for #292

Everything seems working so far.

Question: I'm not completely happy with the way the `text()` tests look. Should we use [snapshot testing](https://facebook.github.io/jest/docs/en/snapshot-testing.html)? Or do you feel these tests are too brittle overall?

Alternatively I could change the component and make it easier to test by adding html classes. 

@schuyler1d  mentioned some good ones to test in #292. Does it make sense to do all those here? It might take me a few weeks. I don't mind keeping the branch up to date with `master`.

- [x] src/components/TexterStats.jsx
- [x] src/components/TopNav.jsx
- [x] src/components/AssignmentSummary.jsx (in #486)
- [ ] src/components/AssignmentTexter.jsx
- [ ] src/components/CampaignBasicsForm.jsx
- [ ] src/components/CampaignContactsForm.jsx

### CODE REVIEW

File: __test__/TexterStats.test.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    })
+  })
+})
+
+
+describe('TexterStats (Dynamic campaign)', () => {
+  it('contains the right text', () => {
+    const stats = shallow(<TexterStats campaign={campaignDynamic} />)
+    expect(stats.text()).toEqual(
+      'Test45 initial messages sentSomeone541 initial messages sent'
```
Comment (jparkrr) [inline __test__/TexterStats.test.js:85]: This is the kind of thing that's weird

Comment (schuyler1d) [reply __test__/TexterStats.test.js:85]: I suppose you could do stats.find(...).text() and evaluate the sub elements but until it gets more sophisticated, I think it's fine

**Comment by schuyler1d** [comment]
We might merge this before you get all the way through, but while it's open, feel free to keep adding here.

Thanks so much, this will help enormously over time as we build up the suite.

File: __test__/TexterStats.test.js (hunk 1 above)
Comment (shakalee14) [reply __test__/TexterStats.test.js:85]: Agree with  @schuyler1d

Comment (jparkrr) [reply __test__/TexterStats.test.js:85]: Thanks for the feedback!

**Comment by jparkrr** [comment]
I found this which talks about ways to test Apollo components https://medium.com/@ferdingler/testing-apollo-react-components-a7618d8a9e7a

Basically they mostly test the unwrapped component (so the component file will have to change slightly), which sounds good to me.

Any thoughts?

**Comment by schuyler1d** [comment]
The pattern that we have for [another project](https://github.com/MoveOnOrg/mop-frontend/blob/master/src/components/signature-add-form.js#L453) is putting at the bottom:
`export const WrappedComponent = Banana` at the bottom (and then in the test do `import { WrappedComponent as Banana } from 'lib/banana'`) (to use the article's example).  If that works, then let's do that.  Otherwise, do what works, and it's totally good and proper to update a component to make it more testable.

**Pull request merged**